include/circt/Dialect/FIRRTL/FIRRTLExpressions.td

```diff
@@ -1176,6 +1176,26 @@ def DoubleConstantOp : FIRRTLOp<"double", [Pure, ConstantLike]> {
   let assemblyFormat = "$value attr-dict";
 }
 
+class IntegerBinaryPrimOp<string mnemonic, list<Trait> traits = []> :
+   BinaryPrimOp<mnemonic, FIntegerType, FIntegerType, FIntegerType,
+                [Pure] # traits> {
+  let inferType = "impl::inferIntegerBinaryPrimResult";
+}
+
+def IntegerAddOp : IntegerBinaryPrimOp<"integer.add", [Commutative]> {
```

Comment:
Is it reasonable to add `SameTypeOperands, SameOperandsAndResultType`?

Reply:
Yes; I think all of these will have `lhs`, `rhs`, and `result` as FIntegerType. I thought about adding these traits, but then I didn't go for it because we already explicitly specified those concrete types in our use of `BinaryPrimOp`. I think we still have to define `inferType` either way. Did I miss something that those traits would help with here?

Reply:
Ah I see. I understand BinaryPrimOp already implements InferType. I thought adding those traits would help removing operand and result types from mlir text format but it seems not to compose well here. Thank you for trying it out.